Libraries/Image/Image.ios.js

```diff
@@ -42,6 +42,22 @@ function getSize(
   );
 }
 
+function getSizeWithHeaders(
+  uri: string,
+  headers: {[string]: string},
+  success: (width: number, height: number) => void,
+  failure?: (error: any) => void,
+) {
+  ImageViewManager.getSizeWithHeaders(
```

Comment:
Can you make this API a promise based API? See the PR here: https://github.com/facebook/react-native/pull/18875/files

We should try to adopt this pattern for new APIs since we cannot change the old one.